cupy/lib/_routines_poly.py

```diff
@@ -157,6 +158,31 @@ def _polypow(x, n):
     return cupy.convolve(x, _polypow(cupy.convolve(x, x), (n - 1) // 2))
 
 
+def _polypow(x, n):
+    if n == 0:
+        return 1
+    if n == 1:
+        return x
+
+    method = cupy._math.misc._choose_conv_method(x, x, 'full')
+
+    if method == 'direct':
+        return _polypow_direct(x, n)
+    elif method == 'fft':
+        if x.dtype.kind == 'c':
+            fft, ifft = cupy.fft.fft, cupy.fft.ifft
+        else:
+            fft, ifft = cupy.fft.rfft, cupy.fft.irfft
+        out_size = (x.size - 1) * n + 1
+        size = cupyx.scipy.fft.next_fast_len(out_size)
+        fx = fft(x, size)
+        fy = cupy.power(fx, n, fx)
+        y = ifft(fy, size)
```

Comment:
Naive algorithm that repeats convolutions `n`-times is

```py
out = [0, ..., 0, 1]
for i in range(n):
    fx = fft(x)
    fout = fft(out)
    fout = elementwise_multiply(fout, fx)
    out = ifft(fout)
```

The above algorithm is equivalent with following one because `fft(ifft(fout))` is equal to `fout`.

```py
out = [0, ..., 0, 1]
fout = fft(out)  # [1, ..., 1]
fx = fft(x)
for i in range(n):
    fout = elementwise_multiply(fout, fx)
out = ifft(fout)
```

So, the final `fout` can be obtained by calculating elementwise_power.